### PR TITLE
I174 process creator

### PR DESCRIPTION
**Pull request by gypsephi**
implemented processor creator as mentioned in #174.
setting up a new process from user mode requires quite a lot invocations and sophisticated cap handling...

is there a method for testing whether a cap entry is existing in the cap map?
useful when copying caps for scheduling contexts and interrupt controls!

Compiles without warnings.
Tested with qemu and IHK.

### CODE REVIEW

**Comment by kubanrob** [comment]
From a first glance, we should talk about the implementation in context of #110 with @rottaran. Other than that, I think I will need a bit more time to review/understand this PR.

**Comment by gypsephi** [comment]
update to master.
tested with qemu and IHK

File: kernel/runtime/process/runtime/process.hh [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+    TEST(res);
+    capAlloc.freeEmpty(pm16.cap());
+
+    res = myCS.move(pl, pm17.cap(), max_cap_depth, cs.cap(), pCapAlloc(), max_cap_depth).wait();
+    TEST(res);
+    capAlloc.freeEmpty(pm17.cap());
+
+    res = myCS.move(pl, pm4.cap(), max_cap_depth, cs.cap(), init::PML4, max_cap_depth).wait();
+    TEST(res);
+    capAlloc.freeEmpty(pm4.cap());
```
Comment (kubanrob) [inline kernel/runtime/process/runtime/process.hh:322]: This feels like a bit too much code duplication. Can we put the caps into some kind of container (array?) and just loop?

Makes the code before that easier to read also.

**Review by kubanrob — APPROVED** [review]

File: kernel/runtime/process/runtime/process.hh (hunk 1 above)
Comment (gypsephi) [reply kernel/runtime/process/runtime/process.hh:322]: I think I will leave it like this for now.
This part of code will be obsolete and be removed when the mmap function got fixed and page tables are created automatically.

**Pull request merged**